schemars/src/json_schema_impls/camino.rs

```diff
@@ -0,0 +1,36 @@
+use crate::gen::SchemaGenerator;
+use crate::schema::*;
+use crate::JsonSchema;
+use camino::{Utf8Path, Utf8PathBuf};
+
+impl JsonSchema for Utf8Path {
+    no_ref_schema!();
+
+    fn schema_name() -> String {
+        "String".to_owned()
+    }
+
+    fn json_schema(_: &mut SchemaGenerator) -> Schema {
+        SchemaObject {
+            instance_type: Some(InstanceType::String.into()),
+            ..Default::default()
+        }
+            .into()
+    }
+}
+
+impl JsonSchema for Utf8PathBuf {
+    no_ref_schema!();
+
+    fn schema_name() -> String {
+        "String".to_owned()
```

Comment:
So hmmmm...

I _think_ this shouldn't be `String`, I believe -- it should be something more specific like `Utf8PathBuf`.

This goes into the `format` field in the OpenAPI and JSON schema spec. There are some defined formats, but otherwise it's an open universe (based on https://swagger.io/docs/specification/data-models/data-types/ and https://json-schema.org/understanding-json-schema/reference/string#format).

But maybe this should be exposed as a custom field on the side, e.g. `x-rust-type`. @ahl do you have thoughts?